src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -33,10 +33,28 @@ type RedpandaResourceRequirements struct {
 // RedpandaCPU returns a copy of the rounded value for Redpanda CPU
 //
 // If it's not explicitly set, the Request.Cpu is used.
+//
+// The value returned is:
+// * Is rounded up to an integer.
+// * Is limited by 2Gi per core if requests.memory is set.
+//
+// Example:
+//    in: minimum requirement per core, 2GB
+//    in: Requests.Memory, 16GB
+//    => maxAllowedCores = 8
+//    if requestedCores == 8, set smp = 8 (with 2GB per core)
+//    if requestedCores == 4, set smp = 4 (with 4GB per core)
 func (r *RedpandaResourceRequirements) RedpandaCPU() *resource.Quantity {
```

Comment:
May return an `int` to signal that it computes the discrete cores (all callers get the `.Value()` from it)

Reply:
Actually, this makes it awkward, because I can't do a nil check. It's possible that no `repanda.cpu` or `requrests.cpu` have been set.